src/api/order/services/getOrderList.js

```diff
@@ -0,0 +1,33 @@
+import sendPaginatedOrderList from '../../../utils/sendPaginatedOrderList';
+import { USER_ROLES } from '../../../utils/constants';
+import { sendErrorResponse } from '../../../utils/responseHelpers';
+
+function getOrderList(req, res) {
+  try {
+    const {
+      limit, pageNo, status, dateFrom, dateTo,
+    } = req.body;
+    const { user } = req;
+    let filter = {};
+
+    if (status && user.role === USER_ROLES.user) {
```

Comment:
Why are you checking this 2 conditions together, if user is calling and status is not sent, you will give to user other users' orders, is it okay?

Reply:
You have to check if user.role is user, then add in filter, no matter what are other conditions.